### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by alanluqman**
## Description 
In this PR, I have Created custom methods for models , and created association between models. 
add two user 
add four posts 
add six comments

first user

last user

all post

all comments

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts, class_name: 'Post'
+  has_many :comments, class_name: 'Comment'
+  has_many :likes, class_name: 'Like'
```
Comment (Nemwel-Boniface) [inline app/models/user.rb]: - [x] When setting up the associations for your models, it is a good thing to also include the foreign keys for them as well. eg ```  has_many :posts, foreign_key: :author_id```

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :comments, class_name: 'Comment'
+  has_many :likes, class_name: 'Like'
+
+  after_save :update_post_counter
```
Comment (Nemwel-Boniface) [inline app/models/post.rb:6]: Good job using callbacks! :+1:

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @alanluqman     :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alanluqman** [comment]
Thanks @Nemwel-Boniface  for your review, I did your requires

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @alanluqman,

✔️ Good documentation of Active Record associations

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**